### PR TITLE
disable knonnectivity by default

### LINKED ISSUE #102904: "dial timeout" flakes in e2e tests with konnectivity proxy enabled
#### What happened:

various e2e tests are flaking with "dial timeout" errors since June 7th, when the konnectivity network proxy was enabled by default in 

https://storage.googleapis.com/k8s-gubernator/triage/index.html?pr=1&text=dial%20timeout

image

All the failures I saw in https://testgrid.k8s.io/sig-api-machinery-network-proxy#ci-kubernetes-e2e-gci-gce-network-proxy-grpc&width=20&sort-by-failures= were "dial timeout" errors

cc @cheftako @jkh52 

/kind bug
/kind flake

### PR DESCRIPTION
**Pull request by BenTheElder**
this is introducing far too much flakiness into CI. it can be re-enabled in the future and specific test jobs can opt *in*.

#### What type of PR is this?

/kind flake

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/102904

#### Special notes for your reviewer:

See: https://github.com/kubernetes/kubernetes/issues/102904, https://kubernetes.slack.com/archives/C09QZ4DQB/p1623960899173900

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster/gce/config-test.sh [hunk 1]
```diff
@@ -44,10 +44,10 @@ WINDOWS_NODE_LABELS=${WINDOWS_NODE_LABELS:-}
 NODE_LOCAL_SSDS_EPHEMERAL=${NODE_LOCAL_SSDS_EPHEMERAL:-}
 # Turning GRPC based Konnectivity testing on id advance of
 # removing the SSHTunnel code.
-export KUBE_ENABLE_EGRESS_VIA_KONNECTIVITY_SERVICE=true
-export PREPARE_KONNECTIVITY_SERVICE="${KUBE_ENABLE_KONNECTIVITY_SERVICE:-true}"
-export EGRESS_VIA_KONNECTIVITY="${KUBE_ENABLE_KONNECTIVITY_SERVICE:-true}"
-export RUN_KONNECTIVITY_PODS="${KUBE_ENABLE_KONNECTIVITY_SERVICE:-true}"
+export KUBE_ENABLE_EGRESS_VIA_KONNECTIVITY_SERVICE="${KUBE_ENABLE_EGRESS_VIA_KONNECTIVITY_SERVICE:-false}"
```
Comment (BenTheElder) [inline cluster/gce/config-test.sh:47]: CI should opt into this in appropriate jobs.
I'm not clear why we added this _and_ `EGRESS_VIA_KONNECTIVITY` yet, while this one was not overridable.

**Comment by BenTheElder** [comment]
xref: https://github.com/kubernetes/kubernetes/pull/102661
cc @Jefftree

File: cluster/gce/config-test.sh (hunk 1 above)
Comment (liggitt) [reply cluster/gce/config-test.sh:47]: I have mixed feelings on this... this has to be a reasonable default configuration before 1.22 is released, so enabling by default makes sense to me. That said, https://github.com/kubernetes/kubernetes/issues/102904 clearly indicates there's an issue to resolve, so I think it is reasonable to back out the "on by default" change until that is resolved.

Comment (aojea) [reply cluster/gce/config-test.sh:47]: I think that it stills need to mature a bit more for being default https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/239

**Comment by BenTheElder** [comment]
Since It's not 100% clear what the true root cause is for the mass of flakes in GCE jobs, I'm going to run a number of iterations of CI on this PR and my stock "no real changes" PR for comparison to see if this somewhat objectively does indeed move the needle.

/test pull-kubernetes-e2e-gce-ubuntu-containerd

green: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/102962/pull-kubernetes-e2e-gce-ubuntu-containerd/1405627138219970560/

**Comment by BenTheElder** [comment]
green: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/102962/pull-kubernetes-e2e-gce-ubuntu-containerd/1405761103559921664/
/test pull-kubernetes-e2e-gce-ubuntu-containerd

**Comment by aojea** [comment]
> Since It's not 100% clear what the true root cause is for the mass of flakes in GCE jobs,

it is 99% clear to me, the logs show a lot of re connections on the konnectivity server that matches with the dial timeout errors, we also seen that after disable it in the network policy job it stopped to flake.
LGTM but defer to Jordan

**Comment by BenTheElder** [comment]
so far this PR and the test PR with no changes are producing equivilant results which is not helpful for positively ID-ing this as a fix to anything.

**Comment by aojea** [comment]
> so far this PR and the test PR with no changes are producing equivilant results which is not helpful for positively ID-ing this as a fix to anything.

I can't fight against that XD

/test pull-kubernetes-e2e-gce-ubuntu-containerd

**Comment by liggitt** [comment]
per https://github.com/kubernetes/kubernetes/pull/102962#discussion_r653956118, backing out the default until https://github.com/kubernetes/kubernetes/issues/102904 is plausibly resolved sgtm

**Comment by aojea** [comment]
/lgtm

we have still signal in https://testgrid.k8s.io/sig-api-machinery-network-proxy#Summary&width=20&sort-by-failures=

Maybe I'm using the wrong word in my previous comments, when I refer to maturity I refer to performance and reliability, as in number of request per second, % failed request, tail latencies ... these proxies/tunnel things are very complex, specially this one that uses multiple layers. I didn't see tests exercising requests per seconds and throughput, also there is some open issues related like
https://github.com/kubernetes-sigs/apiserver-network-proxy/issues/236#issue-917308129 


I did a quick test using the client in the repo, and it start to fail fore more than 1k requests in a row https://github.com/kubernetes-sigs/apiserver-network-proxy/pull/239#issuecomment-864618550
I couldn't investigate more , but I see some room to improvement on the packet processing logic without using channels.

**Comment by BenTheElder** [comment]
> we have still signal in https://testgrid.k8s.io/sig-api-machinery-network-proxy#Summary&width=20&sort-by-failures=

That is not signal that the presubmits are broken by this. Specific jobs can opt-out as necessary without changing the default.

>  also there is some open issues related like

I doubt that's related, it's not like typical e2e tests are doing high bandwidth operations.

**Comment by BenTheElder** [comment]
AIUI we have not seen a signal flake on the test PR versus this one still.

I am seeing instead flakes like this https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/102967/pull-kubernetes-e2e-gce-ubuntu-containerd/1406999832492511232/

**Comment by aojea** [comment]
> I doubt that's related, it's not like typical e2e tests are doing high bandwidth operations.

it is not about bw usually, is about request per seconds and latency in these cases

**Comment by BenTheElder** [comment]
Pass rates have changed.

**Pull request closed without merging**